### ISSUE TITLE
Don't use `online` event and `navigator.onLine` properties

### ISSUE THREAD

**Issue by marco-c** [mozilla/serviceworker-cookbook#151]
Filed as a follow-up of #138.

We should better understand their drawbacks and possibly stop using them.

https://bugzilla.mozilla.org/show_bug.cgi?id=1228313
https://bugzilla.mozilla.org/show_bug.cgi?id=1228309

**Comment by darkwing** [comment]
How shall we replace it?  Just try to hit an arbitrary URL, maybe on the serviceworke.rs domain, to see if it reaches?

**Comment by marco-c** [comment]
Yes, we can ping a domain that's most probably online (e.g. www.mozilla.org).

Let's first understand what's the actual status of the event and properties (are they supported by all browsers? Are they consistent on different OSes?).
